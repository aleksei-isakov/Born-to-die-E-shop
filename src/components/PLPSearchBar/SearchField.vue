<template>
  <div class="search-field">
    <input
      v-model="inputValue"
      class="search-field__input"
      :placeholder="placeHolder"
      @keypress.enter="onClickSearch"
    />
    <BaseButton class="search-field__btn" @click="onClickSearch">
      <BaseCustomIcon :icon="searchIcon" :width="searchIconWidth" />
    </BaseButton>
  </div>
</template>

<script>
import { BaseButton, BaseCustomIcon } from '@/base_components';

export default {
  name: 'SearchField',

  components: {
    BaseButton,
    BaseCustomIcon
  },

  props: {
    placeHolder: {
      type: String,
      default: '',
      required: true
    }
  },

  data() {
    return {
      searchIcon: 'search_icon',
      searchIconWidth: '50%',
      inputValue: ''
    };
  },

  methods: {
    onClickSearch() {
      this.$emit('search', this.inputValue.trim());
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.search-field {
  display: flex;
  justify-content: center;
  flex-grow: 1;
  height: 7vh;
  @media screen and (max-width: $mobile-size) {
    width: 100%;
  }
}

.search-field__input {
  width: 100%;
  border: 1px solid rgb(211, 210, 210);
  border-radius: 5px 0 0 5px;
  display: flex;
  justify-content: flex-end;
  padding-left: 10px;
}

.search-field__btn {
  width: 5vw;
  min-width: 50px;
  height: 100%;
  border: none;
  background-color: $primary;
  border-radius: 0 5px 5px 0;
  cursor: pointer;
}
</style>
