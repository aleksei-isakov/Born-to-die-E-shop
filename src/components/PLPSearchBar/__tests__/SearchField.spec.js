import SearchField from '../SearchField.vue';
import { mount } from '@vue/test-utils';

let wrapper;
const inputValue = 'that is my input!';

beforeEach(() => {
  wrapper = mount(SearchField, {
    propsData: {
      placeHolder: 'input some value'
    },

    data() {
      return { inputValue: inputValue };
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});
describe('SearchField', () => {
  it('should emit search event', () => {
    wrapper.vm.onClickSearch();
    expect(wrapper.emitted().search).toEqual([[inputValue]]);
  });

  it('should render correctly and match valid snapshot', () => {
    const wrapper = mount(SearchField, {
      propsData: {
        placeHolder: 'input some value'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
