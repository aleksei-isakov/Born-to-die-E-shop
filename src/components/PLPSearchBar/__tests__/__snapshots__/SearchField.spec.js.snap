// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField renders correctly 1`] = `
<div
  class="search-field"
>
  <input
    class="search-field__input"
    placeholder="input some value"
  />
   
  <button
    class="search-field__btn"
  >
    <img
      alt="search-icon"
      src=""
    />
  </button>
</div>
`;
